app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -25,6 +25,12 @@ module SmartProxyExtensions
         before_create :associate_lifecycle_environments
         attr_accessible :lifecycle_environment_ids
 
+        has_many :containers,
+                 :class_name => "Container",
+                 :foreign_key => :capsule_id,
+                 :inverse_of => :capsule,
+                 :dependent => :nullify
```

Comment:
Yeah its a tricky case there.. It should only affect if we docker pull that image every time before powering in the container. I think we pull only before create container the first time. May be a better approach is to some how let them migrate to a different smart proxy for this container if they want to delete...